sap/src/test/java/org/frankframework/extensions/sap/GlobalListItemTest.java

```diff
@@ -21,7 +21,7 @@ public void testSapSystemIsRegistered() {
 		sapSystem1.setName("sapSystem1");
 
 		//Act
-		sapSystem1.registerItem(null);
+		GlobalListItem.registerItem(sapSystem1);
```

Comment:
Misschien dat we deze moeten hernoemen naar SapSystemList oid... de huidige naam is niet beschrijvend. Ik snap ook niet precies wat dit doet, en denk ook dat het naar de sap module toe kan om deze verwarrende spagettie uit core te houden.

Reply:
Ik denk dat het idee was om een generieke "global list" implementatie te maken, dus vandaar dat het zo vaag heet. Deze class wordt alleen binnen de sap module gebruikt, dus het verhuizen lijkt me een hele logische stap. Het hernoemen is dan ook een goed idee.

Iedere class die deze class inherit kan toegevoegd worden aan een static lijst. Dit wordt nu alleen gebruikt om in een configuratie het volgende te doen. Het bevat dus de lijst met geregistreerde sap systems.

```xml
<SapSystems>
  <SapSystem />
  <SapSystem />
</SapSystems>
```

Reply:
Ah ik zie nu dat het verplaatsen van de `GlobalListItem` niet zo makkelijk is. De reden voor al deze gekke code en stomme wijzigingen is omdat de sap module een dependency heeft op core, terwijl core soms ook een class van sap wilt gebruiken terwijl dat niet meer mogelijk is, omdat het dan een circulair dependency wordt (wat niet mag).

Dit verklaart waarom `selfRegisterMethod` ipv `registerMethod` gebruikt werd. De `selfRegisterMethod` zit op `SapSystem` wie in de sap module woont, dus alle sap en core code kan gebruiken, terwijl `registerMethod` op `SapSystems` zit (omdat `SapSystems` aan een configuratie toegevoegd moet kunnen worden). Binnen die methode kan alleen de core module gebruikt worden.